### DISCUSSION TITLE
Does faker-js/faker support typescript?

### DISCUSSION THREAD

**Discussion by lalehkoupa** [faker-js/faker#1640] [Q&A] [answered]
Hi,
I used to use the faker library in my typescript project, now I have used faker-js/faker and did install that but apparently, it doesn't support type script as I could not install @type/faker-js/faker,
at the moment I added .d.ts file in the root of my project with "declare module "@faker-js/faker", I am wondering if it is the best solution.
thanks,

**Comment by ST-DDT** [comment, marked as answer]
> I could not install @type/faker-js/faker,

You dont have to. The type definitions are part of the main package since v6.0.0.

**Reply by lalehkoupa** [reply]
Thanks for your reply, I just installed the package yesterday and the version is "@faker-js/faker": "^7.6.0",
but I 've got this compile error: 


error TS7016: Could not find a declaration file for module '@faker-js/faker'. '/app/node_modules/@faker-js/faker/dist/cjs/index.js' implicitly has an 'any' type.
  Try `npm install @types/faker-js__faker` if it exists or add a new declaration (.d.ts) file containing `declare module '@faker-js/faker';

1 import { faker } from "@faker-js/faker";
